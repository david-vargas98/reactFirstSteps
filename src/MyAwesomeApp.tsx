import type { CSSProperties } from "react";

const firstName = "Edgar";
const lastName = "Vargas";

const myFavoriteGames = ['Halo', 'Ninja Gaiden', 'Doom'];
const isActive = true;

const address = {
    zipCode: 'ABC-123',
    country: 'Canada'
}

const myStyles: CSSProperties = {
    backgroundColor: 'red',
    borderRadius: 20,
    padding: 10,
    marginTop: 50
}

export function MyAwesomeApp() {
    return (
        <>
            <h1>{firstName}</h1>
            <h3>{lastName}</h3>

            <p>{myFavoriteGames.join(', ')}</p>

            <h1>{isActive ? 'Active' : 'Not active'}</h1>

            <p style={myStyles}> {JSON.stringify(address)} </p>
        </>
    );
}
