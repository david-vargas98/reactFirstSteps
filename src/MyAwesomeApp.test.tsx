import { describe, expect, test } from "vitest"; // function from vitest to test non-react functionalities
import { render, screen } from '@testing-library/react' // the 'testing library' needed to test the components

import { MyAwesomeApp } from "./MyAwesomeApp"; // my component to be tested

describe('MyAwesomeApp', () => {
    test('should render firstName and lastName', () => {
        const { container } = render(<MyAwesomeApp />); // we use container to evaluate initial state
        //screen.debug(); // it allows to evaluate through constant changes (re-renders) made from the component

        const h1 = container.querySelector('h1');
        const h3 = container.querySelector('h3');

        expect(h1?.innerHTML).toContain('Edgar'); // toContain checks if the 'Edgar' is contained into 'h1' independently if there's a space in between
        expect(h3?.innerHTML).toContain('Vargas');

        //const {container} = render(<MyAwesomeApp />);

        //console.log(container.innerHTML);
    });
    test('should render firstName and lastName - screen', () => {
        render(<MyAwesomeApp />);
        screen.debug();

        // const h1 = screen.getByRole('heading', {
        //     level: 1
        // }); // if we want to get the first h1 we use this syntax

        const h1 = screen.getByTestId('first-name-title'); // we evaluate the element by using a 'data-testid' property

        console.log(h1.innerHTML);
    });

    // getting use of snapshots: use them if the component isn't gonna change anymore
    test('should match snapshot', () => {
        const { container } = render(<MyAwesomeApp />);

        expect(container).toMatchSnapshot(); // generates a snapshot to compare with
    });

    // using attributes such as 'data-testid' is not a wise idea, since they're prone to change
    // however, is good to know their existence
    test('should match snapshot', () => {
        // we don't have any native snapshot method to get the html template, so we use 'data-testid'
        render(<MyAwesomeApp />);

        expect(screen.getByTestId('div-app')).toMatchSnapshot(); 
    });
});