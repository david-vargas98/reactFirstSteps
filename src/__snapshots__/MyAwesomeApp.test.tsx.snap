// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
      Edgar
    </h1>
    <h3>
      Vargas
    </h3>
    <p>
      Halo, Ninja Gaiden, Doom
    </p>
    <h1>
      Active
    </h1>
    <p
      style="background-color: red; border-radius: 20px; padding: 10px; margin-top: 50px;"
    >
       
      {"zipCode":"ABC-123","country":"Canada"}
       
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snapshot 2`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
    Edgar
  </h1>
  <h3>
    Vargas
  </h3>
  <p>
    Halo, Ninja Gaiden, Doom
  </p>
  <h1>
    Active
  </h1>
  <p
    style="background-color: red; border-radius: 20px; padding: 10px; margin-top: 50px;"
  >
     
    {"zipCode":"ABC-123","country":"Canada"}
     
  </p>
</div>
`;
